src/main/java/ai/apptuit/metrics/jinsight/ConfigService.java

```diff
@@ -184,9 +184,44 @@ private static Properties loadProperties(File configFilePath) throws IOException
     try (InputStream inputStream = new BufferedInputStream(new FileInputStream(configFilePath))) {
       config.load(inputStream);
     }
+    // override with system properties, if present
+    config.putAll(loadSystemProperties());
     return config;
   }
 
+  private static Map<String, Object> loadSystemProperties() {
+    Map<String, Object> systemProperties = new HashMap<>();
+    String reporter = getProperty(REPORTER_PROPERTY_NAME);
+    if(reporter != null && !reporter.equals("")){
+      systemProperties.put(REPORTER_PROPERTY_NAME, reporter);
+    }
+
+    String accessToken = getProperty(ACCESS_TOKEN_PROPERTY_NAME);
+    if(accessToken != null && !accessToken.equals("")){
+      systemProperties.put(ACCESS_TOKEN_PROPERTY_NAME, accessToken);
+    }
+
+    String apiEndpoint = getProperty(API_ENDPOINT_PROPERTY_NAME);
+    if(apiEndpoint != null && !apiEndpoint.equals("")){
+      systemProperties.put(API_ENDPOINT_PROPERTY_NAME, apiEndpoint);
+    }
+
+    String globalTags = getProperty(GLOBAL_TAGS_PROPERTY_NAME);
```

Comment:
System prop name should be jinsight.global_tags (even though the property name is just global_tags)